### PR TITLE
Small refactor to ensure 'to_map_datapipe' is generated as a method in IterDataPipe's pyi interface

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #329
* #328
* #327
* __->__ #326
* #325

Previously, `.to_map_datapipe` didn't show up as a hint due to how the file crawling/parsing work in `gen_pyi.py`. This fixes it.

Differential Revision: [D35979907](https://our.internmc.facebook.com/intern/diff/D35979907)

### CODE REVIEW

**Review by ejguan — APPROVED** [review]
LGTM Thank you!

File: torchdata/datapipes/iter/util/converter.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
     dill.extend(use_dill=False)
 
 
+# @functional_datapipe("to_map_datapipe")  # This line must be kept for .pyi signature parser
```
[the remark this answers is not in the document]
Comment (NivekT) [reply torchdata/datapipes/iter/util/converter.py:20]: Agreed. I think this is acceptable for now but definitely not the best solution.

File: torchdata/datapipes/iter/util/converter.py [hunk 2]
```diff
@@ -67,6 +68,12 @@ def __getitem__(self, index):
             self._load_map()
         return self._map[index]  # type: ignore[index]
 
+    def __iter__(self):
+        if self._map is None:
+            self._load_map()
+        else:
+            yield from self._map.values()
+
```
Comment (ejguan) [inline torchdata/datapipes/iter/util/converter.py:76]: I am sorry that I got confused by these two PRs. Why do we need `__iter__` for this `MapDataPipe`?

Comment (NivekT) [reply torchdata/datapipes/iter/util/converter.py:76]: It is not needed but without it you can't iterate over it. I consider it to be a nice-to-have.

```python
from torchdata.datapipes.iter import IterableWrapper
from torchdata.datapipes.map import IterToMapConverter

dp = IterableWrapper([(i, i) for i in range(10)])
dp = IterToMapConverter(dp)
for i in iter(dp):
    print(i)
```

In the end, you will get `KeyError: 10.` if `__iter__` isn't there.

Comment (ejguan) [reply torchdata/datapipes/iter/util/converter.py:76]: IIRC, You iterate it if you raise proper Error
```py
def __getitem__(self, index):
    try:
        do your stuff
    except KeyError:
        raise IndexError

```

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35979907).

**Pull request closed without merging**